### PR TITLE
Nexus: Add L2 generation functionality

### PR DESCRIPTION
**Pull request by mcbennet**
## Proposed changes

The following PR adds L2 functionality to the GaussianPP class within pseudopotential.py.
These additions only expand the GuassianPP class and do not change any existing functionality.

## What type(s) of changes does this code introduce?
_Delete the items that do not apply_

- New feature

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?

Laptop

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes. This PR is up to date with current the current state of 'develop'
- N/A. Code added or changed in the PR has been clang-formatted
- No. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- No. Documentation has been added (if appropriate)

### CODE REVIEW

File: nexus/lib/pseudopotential.py [hunk 1]
```diff
@@ -53,6 +53,7 @@
 from physical_system import PhysicalSystem
 from plotting import *
 from debug import *
+from testing import *
```
Comment (jtkrogel) [inline nexus/lib/pseudopotential.py:56]: Remove testing import

File: nexus/lib/pseudopotential.py [hunk 2]
```diff
@@ -1956,6 +1971,496 @@ def ppconvert(self,outfile,ref,extra=None):
         execute(command,verbose=True)
         os.system('rm '+tmpfile)
     #end def ppconvert
+
+
+    # test needed
+    def append_to_component(self,l,coeff,expon,rpow):
```
Comment (jtkrogel) [inline nexus/lib/pseudopotential.py:1977]: Add a 1-2 sentence docstring very briefly describing what this function does.  If the role of this function is simply to be called by another, note that here along with the name of the other function.  Please do the same for the other functions below.

**Review by jtkrogel — CHANGES_REQUESTED** [review]
@mcbennet after addressing the comments, please post here the output from nxs-test.

**Comment by mcbennet** [comment]
> @mcbennet after addressing the comments, please post here the output from nxs-test.

The structure test is the only failure when I run nxs-test. Attached is the output of `nxs-test -R structure --verbose
[nxs_structure_test.txt](https://github.com/QMCPACK/qmcpack/files/6283094/nxs_structure_test.txt)
`

**Comment by jtkrogel** [comment]
Please update line 1645 in test_structure.py to read `assert(testing.check_object_eq(res,ref,atol=1e-12))` and let me know if this fixes things.

**Comment by mcbennet** [comment]
> Please update line 1645 in test_structure.py to read `assert(testing.check_object_eq(res,ref,atol=1e-12))` and let me know if this fixes things.

Thank you. The structure test now passes.

**Review by jtkrogel — APPROVED** [review]
LGTM

**Review by ye-luo — APPROVED** [review]

**Comment by prckent** [comment]
OK to test

**Pull request merged**